rye.go

```diff
@@ -131,6 +132,10 @@ func (m *MWHandler) do(w http.ResponseWriter, r *http.Request, handler Handler)
 			}()
 		}
 
+		if resp != nil && resp.StatusCode > 0 {
+			statusCode = strconv.Itoa(resp.StatusCode)
+		}
```

Comment:
I think you can move this inside the `if resp != nil` block above and avoid having to check for nil again, as well as the changes to set the `resp` var separately.
I think you can also remove 
```
statusCode = strconv.Itoa(resp.StatusCode)
```
above `WriteJSONStatus`

Reply:
Thanks for looking at this! I was working with Cale on this on Friday and he suggested having it outside that first `if resp != nil` block. The purpose is to pass something like this elsewhere: 
```
&rye.Response{
  StopExecution: true,
  StatusCode: 404,
}
```
but still process the StatusCode for statsd. The use case is scenarios where we actually don't want Rye to write out the response for us (custom formatted errors, for instance), in which case we want StopExecution to actually skip all that other stuff. But we _do_ still want the response code to get tracked in statsd.